rust/src/lib.rs

```diff
@@ -355,17 +500,68 @@ pub unsafe extern "C" fn zcashlc_create_account(
                 }
             })?;
 
-        db_data
+        let (account_id, usk) = db_data
             .create_account(&seed, birthday)
-            .map(|(account, usk)| {
-                let encoded = usk.to_bytes(Era::Orchard);
-                Box::into_raw(Box::new(FFIBinaryKey::new(account, encoded)))
-            })
-            .map_err(|e| anyhow!("Error while initializing accounts: {}", e))
+            .map_err(|e| anyhow!("Error while initializing accounts: {}", e))?;
+
+        let account = db_data.get_account(account_id)?.expect("just created");
+        let account_index = match account.kind() {
+            AccountKind::Derived { account_index, .. } => account_index,
+            AccountKind::Imported => unreachable!("just created"),
+        };
+
+        let encoded = usk.to_bytes(Era::Orchard);
+        Ok(Box::into_raw(Box::new(FFIBinaryKey::new(
+            account_index,
+            encoded,
+        ))))
     });
     unwrap_exc_or_null(res)
 }
 
+/// Checks whether the given seed is relevant to any of the accounts in the wallet.
+///
+/// Returns:
+/// - `1` for `Ok(true)`.
+/// - `0` for `Ok(false)`.
+/// - `-1` for `Err(_)`.
+#[no_mangle]
+pub unsafe extern "C" fn zcashlc_is_seed_relevant_to_wallet(
+    db_data: *const u8,
+    db_data_len: usize,
+    seed: *const u8,
+    seed_len: usize,
+    network_id: u32,
+) -> i8 {
+    let res = catch_panic(|| {
+        let network = parse_network(network_id)?;
+        let db_data = unsafe { wallet_db(db_data, db_data_len, network)? };
+        let seed = Secret::new((unsafe { slice::from_raw_parts(seed, seed_len) }).to_vec());
+
+        for account_id in db_data.get_account_ids()? {
+            match db_data.validate_seed(account_id, &seed) {
+                // The seed is relevant to this account. No need to check any others.
+                Ok(true) => return Ok(1),
+                // We know by dead reckoning that this account ID exists in the wallet, so
```

Comment:
Not sure what "by dead reckoning" means here.
```suggestion
                // We're looping over account IDs that we know exist in the wallet, so
```